packages/strapi-plugin-upload/services/Upload.js

```diff
@@ -90,7 +90,7 @@ module.exports = {
   async enhanceFile(file, fileInfo = {}, metas = {}) {
     let readBuffer;
     try {
-      readBuffer = await util.promisify(fs.readFile)(file.path);
+      readBuffer = await util.promisify(fs.readFile)(file && file.path || file);
```

Comment:
@alexandrebodin please ask / mention the Strapi developer who wrote this function, a quick git blame shows that you, @petersg83 & @soupette worked on this before..

Reply:
Well, both Files and Buffers need to be passed to fs.readFile as this function works with both by default, see https://nodejs.org/api/fs.html#fs_fs_readfile_path_options_callback

So removing fs.readFile when receiving a Buffer will probably break this whole function as far as I can tell.  Please see the following topics in the community forum related to this issue:
- https://forum.strapi.io/t/need-help-with-reviving-strapi-plugin-import-content/3937
- https://forum.strapi.io/t/upload-image-url/3484/4
- https://forum.strapi.io/t/strapi-upload-service-cannot-handle-buffers/4830

Reply:
From what you are sharing you are not using the Strapi upload API but calling this function programmatically right ? 

What does your Buffer contains exactly ? the `fs.readFile`  expects a file path. Indeed it can be a Buffer but that's still going to try to read the file at the path e.g `fs.readFile(Buffer.from('/tmp/tmpfile.png`)). 

I thought you were trying to pass a buffer with the file content directly :)

can you please share some code snippets here of what you are trying to do in details :)

Reply:
Oh, and if you pass a Buffer to fs.readFile, it will accept a bytestring array (contained in the `buffer`attribute) and treat it as a file, without a path attribute.  This is exactly what I saw during debugging: the Buffer object has not path, but a buffer, containing the (streaming) file contents of the remote file.  So correct me if I'm wrong, but your previous statement that `fs.readFile` expects a file path is not true as it does accept Buffers containing the filedata as a bytearray, without the file being ever persisted on the local fs.

Reply:
Hi Alexandre,

upon further testing I noticed that I'm getting a new error now, and following your assumptions I managed to solve it similar to your approach: if the passed file is an object containing a buffer property, it works as you mentioned before by simply setting the readBuffer to the buffer property of the object.

I'll update the code in my fork to reflect these changes here as well.

Reply:
After more testing it seems your suggestion to pass the buffer straight to the readBuffer variable is the solution indeed!  I've updated my fork accordingly.